# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../dovecot/dovecot-2.2-0005-f33bacb03cc2.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---


# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1366116845 -10800
# Node ID f33bacb03cc22e02ca2d11d527d8dc0d99214486
# Parent  9a5b132b6832137c5ebedb34001354267b41208e
maildir: Fixed crash in some rare situations.

diff -r 9a5b132b6832 -r f33bacb03cc2 src/lib-storage/index/maildir/maildir-util.c
--- a/src/lib-storage/index/maildir/maildir-util.c	Mon Apr 15 21:40:35 2013 +0300
+++ b/src/lib-storage/index/maildir/maildir-util.c	Tue Apr 16 15:54:05 2013 +0300
@@ -87,6 +87,9 @@
 		fname = maildir_filename_guess(mbox, uid, fname,
 					       &flags, &have_flags);
 	}
+	/* make a copy, just in case callback refreshes uidlist and
+	   the pointer becomes invalid. */
+	fname = t_strdup(fname);
 
 	ret = 0;
 	if ((flags & MAILDIR_UIDLIST_REC_FLAG_NEW_DIR) != 0) {

